src/databricks/labs/dqx/rule.py

```diff
@@ -39,20 +39,26 @@ class ExtraParams:
 @dataclass(frozen=True)
 class DQRule:
     """Class to represent a data quality rule consisting of following fields:
-    * `check` - Column expression to evaluate. This expression should return string value if it's evaluated to true -
-    it will be used as an error/warning message, or `null` if it's evaluated to `false`
+    * `check_func` - check function to be applied
+    * `col_name` - column name to which the given check function should be applied
     * `name` - optional name that will be given to a resulting column. Autogenerated if not provided
     * `criticality` (optional) - possible values are `error` (critical problems), and `warn` (potential problems)
+    * `filter` (optional) - filter expression to apply the check only to the rows that satisfy the condition
+    * `check_func_args` (optional) - non-keyword / positional arguments for the check function after the col_name
+    * `check_func_kwargs` (optional) - keyword /named arguments for the check function after the col_name
     """
 
-    check: Column
+    check_func: Callable
+    col_name: str = ""
     name: str = ""
     criticality: str = Criticality.ERROR.value
     filter: str | None = None
+    check_func_args: list[Any] = field(default_factory=list)
+    check_func_kwargs: dict[str, Any] = field(default_factory=dict)
 
     def __post_init__(self):
         # take the name from the alias of the column expression if not provided
```

Comment:
The method should also verify that the provided args and kwargs are actually valid. It should only be possible to create `DQRule` if correct correct args/kwargs are provided for the check_func.

Reply:
Added